sql/src/main/java/io/druid/sql/avatica/DruidStatement.java

```diff
@@ -103,7 +102,7 @@ public DruidStatement(
     this.statementId = statementId;
     this.queryContext = queryContext == null ? ImmutableMap.of() : queryContext;
     this.onClose = Preconditions.checkNotNull(onClose, "onClose");
-    this.yielderOpenCloseExecutor = Execs.singleThreaded("JDBCYielderOpenCloseExecutor");
+    this.yielderOpenCloseExecutor = Execs.singleThreaded(String.format("JDBCYielderOpenCloseExecutor-statement-%d", statementId));
```

Comment:
Should include the connection ID too, since the IDs are not unique across connections.